keps/sig-windows/1122-windows-csi-support/README.md

```diff
@@ -747,33 +756,410 @@ All E2E storage tests covering CSI plugins will be ported to Windows workloads a
+non-trivial computations, excessive access to disks (including increased log
+volume), significant amount of data sent and/or received over network, etc.
+This through this both in small and large cases, again with respect to the
+[supported limits].
+
+[supported limits]: https://git.k8s.io/community//sig-scalability/configs-and-limits/thresholds.md
+-->
+There are two aspects of resource usage in the context of CSI proxy
+
+[a] specific latencies and extra resources consumed as a result of driving operations through CSI plugin using CSI proxy vs. In-tree plugin with Support for Windows:
+
+The preliminary investigation using GCE PD CSI driver on this is available [link](https://docs.google.com/document/d/10aXLjJs8HvloY0zQJaMsOMn0S0AyrVyfh8iC9T7A1yg/edit). There are relatively small extra delay caused by CSI proxy for certain volume operations. There will be future performance investigation and improvement work on this area.
+
+[b] the overall resources associated with running operations (such as list disk/format volumes/etc) associated with CSI plugins vs not running CSI plugins on Windows?
+
+The tests performed on this area has not shown noticeable overhead. Plan to test on larger scale in the future.
+
+### Troubleshooting
+
+<!--
+This section must be completed when targeting beta to a release.
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+-->
+
+CSI Proxy is a thin stateless API layer between CSI node plugins (running on Windows) and Windows OS APIs (surfaced through Win32 or Powershell cmdlets). Troubleshooting of APIs associated with CSI Proxy should be performed using either logs from the CSI plugin or Windows OS event logs."
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+No impact
+
+###### What are other known failure modes?
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+1. CSI Proxy binary is not installed and running
+2. CSI-proxy version and Node Plugin version do not match
+
+The error message from CSI Node plugin will show, for example
+```
+"transport: Error while dialing open \\\\.\\\\pipe\\\\csi-proxy-disk-v1beta2: The system cannot find the file specified."
```

Comment:
Is there a way to expose this (or a similar, identifiable message) in a kubelet log or an event?

Reply:
this error message is shown too in events for Pod which is trying to mount PVC.

Reply:
I will update this with the information